src/averageAges.js

```diff
@@ -38,6 +52,26 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const filteredWomen = people.filter((person) => person.sex === 'f');
+  const filteredMothers = [];
+
+  for (const key of filteredWomen) {
+    const checkCondition = people.some(women => key.name === women.mother);
```

Comment:
Create more meaningful names
```suggestion
    const isMother= people.some(person=> woman.name === person.mother);
```